src/xo-app/navbar/index.js

```diff
@@ -0,0 +1,41 @@
+import Icon from 'icon'
+import React, { Component } from 'react'
+import {
+  connectStore,
+  propTypes
+} from 'utils'
+
+@connectStore([
+  'user'
+])
+@propTypes({
+  selectLang: propTypes.func.isRequired
+})
+export default class Navbar extends Component {
+  handleSelectLang (event) {
+    // FIXME: find a way to reach selectLang to set the app language
+    this.props.selectLang(event.target.value)
```

Comment:
It does not work because the context is not define when the method is called without `this`:

```js
this.handleSelectLang() // Context correctly defined in handleSelectLang().
const handleSelectLang = this.handleSelectLang
handleSelectLang() // Context is not defined in handleSelectLang() because not called with this.
```

You can use the `@autobind` decorator in `utils` to fix that:

```js
@autobind
handleSelectLang (event) {
  // ...
}
```